warehouse/query-core/src/test/java/datawave/query/TestLimitReturnedGroupsToHitTermGroups.java

```diff
@@ -273,6 +273,22 @@ public void testMultipleGroupsWithRegexAndReturnFields() throws Exception {
         runTestQuery(queryString, format.parse("20091231"), format.parse("20150101"), extraParameters, goodResults);
     }
     
+    @Test
+    public void testGroupWithRegexAndMatchesInGroup() throws Exception {
+        Map<String,String> extraParameters = new HashMap<>();
+        extraParameters.put("include.grouping.context", "true");
+        extraParameters.put("hit.list", "true");
+        extraParameters.put("limit.fields", "CANINE=-1,BIRD=-1");
+        extraParameters.put("return.fields", "CANINE,BIRD");
```

Comment:
For the keys, use the QueryOptions or QueryParameters constants instead of creating new strings

Reply:
Good idea.  Since we do this across a bunch of other test classes, I'll do them all at once under a follow-on MR.